README.md

```diff
@@ -58,42 +95,235 @@ import ReactDOM from 'react-dom';
+
+#### API
+
+* `action.loading` - flag of whether action is executing
+  - Type: {boolean}
+  - Default: false
+* `action.error` - flag of whether action has error after executing
+  - Type: {boolean}
+  - Default: false
+* `action.disableLoading` - flag of whether disable action's loading effect, if set to true, views will not rerender when loading status changed
+  - Type: {boolean}
+  - Default: false
+* `store.disableLoading` - flag of whether disable loading effect globally. Note, action's `disableLoading` flag will always takes priority when both are set
+  - Type: {boolean}
+  - Default: false
+
+#### Example
+
+```javascript
+const todos = store.useStore('todos');
+
+useEffect(() => {
+  todos.refresh();
+}, []);
+
+return (
+  <div>
+    {todos.refresh.error ? 
+      todos.refresh.loading ? (
+	    <div>
+	      loading.......
+	    </div>
+      ) : (
+        <div>
+	      error.......
+	     </div>
+      ) 
+    : (
+       <ul>
+         {dataSource.map(({ name }) => (
+           <li>{name}</li>
+         ))}
+       </ul>
+    )}
+  <Loading />
 );
 ```
+## Testing
 
-## 示例
+Because all the state and action are cleanly contained in a plain object, it's easy to write test without mocking.
 
-- Todos：[Sandbox](https://codesandbox.io/s/2017600okp)
+Example:
 
-## 反馈
+```javascript
+describe("todos", () => {
+  test("refresh data success", async () => {
+    await todos.refresh();
+    expect(todos.dataSource).toEqual([
+      {
+        name: "react"
+      },
+      {
+        name: "vue",
+        done: true
+      },
+      {
+        name: "angular"
+      }
+    ]);
+  });
+});
+```
 
-- [issues](https://github.com/alibaba/ice/issues)
+Please refer to the `todos.spec.js` file in the above sandbox for complete reference.
 
-## 参考
+## Best Practice
+
+### Never mutate state outside actions
+
+icestore enforce all the mutations to state can only ouccr in action methods. It will throw error when mutate state directly outside actions(eg. in view component). It will cause your mutation logic hard to trace and imposible to test if let mutation scatterd around view component other than store.
+
+```javascript
+  // store.js
+  export default {
+    inited: false,
+    setInited() {
+      this.inited = true;
+    }
+  }
+  
+  // view.js
+  const todos = useStore('todos');
+  
+  useEffect(() => {
+    // bad
+    todos.inited = true;
+    
+    // good
+    todos.setInited();
+  });
+```
+
+### Divide store as small as possible
+
+By design, `icestore` will trigger the rerender of all the view component subscribed to the store by `useStore` once the state of the store has changed. Which means puting more state in one store may cause more view compoenent to rerender which will affect the performance of app. So from performance aspect, it's advised to categorize your state and put them in indivisual stores. 
```

Comment:
By design, `icestore` will trigger the rerender of all the view components subscribed to the store (by using `useStore`) once the state of the store has changed. 

This means that putting more state in one store may cause more view components to rerender, affecting the overall performance of the application. As such, it is advised to categorize your state and put them in individual stores to improve performance.